### PR TITLE
Allow for optional allow-multiple query params

### PR DESCRIPTION
**Pull request by davidkonigsberg**
Previously, when a query parameter was set to allow-multiple, it would no longer be optional. This PR attempts to fix that.

### CODE REVIEW

File: seed/sdk/trace/src/seed/resources/playlist/client.py [hunk 1]
```diff
@@ -271,7 +271,7 @@ async def get_playlists(
         limit: typing.Optional[int] = None,
         other_field: str,
         multi_line_docs: str,
-        optional_multiple_field: typing.Union[typing.Optional[str], typing.List[str]],
+        optional_multiple_field: typing.Optional[typing.Union[typing.Optional[str], typing.List[str]]] = None,
```
Comment (dsinghvi) [inline seed/sdk/trace/src/seed/resources/playlist/client.py:274]: @davidkonigsberg  i would expect  the first type inside the union here just be a `str`

File: src/fern_python/generators/sdk/client_generator/endpoint_function_generator.py [hunk 2]
```diff
@@ -709,6 +700,30 @@ def _get_header_parameter_name(self, header: ir_types.HttpHeader) -> str:
     def _get_query_parameter_name(self, query_parameter: ir_types.QueryParameter) -> str:
         return query_parameter.name.name.snake_case.safe_name
 
+    def _get_typehint_for_query_param(self, query_parameter: ir_types.QueryParameter, query_parameter_type_hint: AST.TypeHint) -> AST.TypeHint:
+        value_type = query_parameter.value_type.get_as_union()
+        is_optional = value_type.type == "container" and value_type.container.get_as_union().type == "optional"
+        if is_optional and query_parameter.allow_multiple:
+            return AST.TypeHint.optional(AST.TypeHint.union(
+                query_parameter_type_hint,
```
Comment (dsinghvi) [inline src/fern_python/generators/sdk/client_generator/endpoint_function_generator.py:708]: i think this should be the type hint based on the `value_type`

Comment (dsinghvi) [reply src/fern_python/generators/sdk/client_generator/endpoint_function_generator.py:708]: could be `unwrap_optional_type(query_parameter_type_hint)`

Comment (davidkonigsberg) [reply src/fern_python/generators/sdk/client_generator/endpoint_function_generator.py]: Thanks for the hint! Changes made.

**Pull request merged**